mono/mini/debugger-engine.c

```diff
@@ -592,7 +592,25 @@ mono_de_clear_breakpoints_for_domain (MonoDomain *domain)
 static int ss_count;
 
 /* The single step request instance */
-static SingleStepReq *the_ss_req;
+static GPtrArray *the_ss_reqs;
+
+static void
+ss_req_init (void)
+{
+	the_ss_reqs = g_ptr_array_new ();
+}
+
+static void
+ss_req_cleanup (void)
+{
+	mono_loader_lock ();
```

Comment:
`dbg_lock` ?  (I am not sure.  but in `ss_req_acquire` you use `dbg_lock` to protect the array.  Is cleanup different?)

Reply:
Yes, you are right! thanks! :)